File: src/app/model/BookingDetails.ts
import { Passenger } from "./Passenger";

export class BookingDetails{
    id: number=0;
    pnr: string = "";
    //flightNo: number = 0;
    noOfPersons: number = 0;
    amount: number = 0;
    totalAmount: number = 0;
    departureDate: string = "";
    returnDate: string = "";
    //departure: string = "";
    //arrival: string = "";
    email: string = "";
    fromPlace: string = "";
    toPlace: string = "";
    cancelled: boolean=false;
    passengerList: Passenger[]=[];
    flightDetails1: number = 0;
    flightDetails2: number = 0;
    flightSchedule1: number = 0;
    flightSchedule2: number = 0;
    couponId: number = 0;
    seatClass: string = "";
    //schedule related fields
}

File: Java Entity Class/BookingDetails.java
class BookingDetails{
    private int id;
    private String pnr;
    private String flightDetails1; //Reference to FlightDetails 1 for oneway trip
    private String flightDetails2; //Reference to FlightDetails 1 for round trip
    private int noOfPersons;
    private double amount;  // to store with out discount
    private double totalAmount;
    private Date departureDate;
    private Date arrivalDate;
    private String email;
    private String fromPlace;
    private String toPlace;
    private String passengers;  // Reference to Passengers List FK
    private boolean camcelled=false;
}